website/addons/base/__init__.py

```diff
@@ -345,6 +346,11 @@ def deep_url(self):
             return url + '/'
 
     @property
+    def source_url(self):
+        extra = self._metadata_cache['extra']
```

Comment:
Will there be any case where `self._metadata_cache` doesn't contain the key `extra`?

If so, perhaps this would be clearer:

```python
    @property
    def source_url(self):
        try:
            return self._metadata_cache['extra']['source_url']
        except KeyError:
            return None
```

If you're confident that `extra` will always be there, make no changes :)